cdap-ui/cypress/integration/pipeline.plugin.multipleendpoints.spec.ts

```diff
@@ -33,19 +46,75 @@ describe('Pipelines with plugins having more than one endpoints', () => {
       });
     });
   });
-  afterEach(() => {
+
+  after(() => {
     // Delete the pipeline to clean up
     pipelines.forEach((pipeline) => cy.cleanup_pipelines(headers, pipeline));
   });
 
-  it('Should work with union splitter and condition plugins', (done) => {
-    Helpers.deployAndTestPipeline(
+  it('Should upload pipeline with union splitter and condition plugins', () => {
+    cy.visit('/pipelines/ns/default/studio');
+    cy.get('#resource-center-btn').click();
+    cy.get('#create-pipeline-link').click();
+    cy.url().should('include', '/studio');
+    cy.upload_pipeline(
       'union_condition_splitter_pipeline_v1-cdap-data-pipeline.json',
-      unionConditionPipeline,
-      done
+      '#pipeline-import-config-link > input[type="file"]'
     );
+    // This is arbitrary. Right now we don't have a way to determine
+    // if the upgrade check is done. Since this a standalone the assumption
+    // is this won't take more than 10 seconds.
+    cy.wait(10000);
+    // Name pipeline
+    cy.get('.pipeline-name').click();
+    cy.get('#pipeline-name-input')
+      .clear()
+      .type(unionConditionPipeline)
+      .type('{enter}');
   });
-  it('Should work with null splitter plugin', (done) => {
+  it('Should show preview data for multiple outputs for splitter and correct message for conditional', () => {
```

Comment:
Does this test depend on the previous test? If so, can you restructure so it can be run independently?

Reply:
Good call.  I've moved the uploading of the first pipeline to a before hook since both two other tests depend on it, so now the tests do not rely on each other.